src/Handlers/MarkdownHandler.php

```diff
@@ -75,23 +75,21 @@ function ($path) use ($pageData, $layout) {
 
     private function getEscapedMarkdownContent($file)
     {
-        $markdownFileContent = $this->escapePhpOpenTags($file->getContents());
+        $content = str_replace('<?php', "__PHP_OPEN__", $file->getContents());
 
         if ($file->getFullExtension() == 'md') {
-            $markdownFileContent = $this->escapeAtSymbolInMarkdown($markdownFileContent);
+            $content = str_replace(
+                ['{{', '}}', '{!!', '!!}', '@'],
+                ['__OPEN_CURLIES__', '__CLOSE_CURLIES__', '__OPEN_DANGER__', '__CLOSE_DANGER__', '__LONE_AT__'],
+                $content
+            );
         }
 
-        return $markdownFileContent;
-    }
-
-    private function escapePhpOpenTags($content)
-    {
-        return str_replace("<?php", "<{{'?php'}}", $content);
-    }
-
-    private function escapeAtSymbolInMarkdown($content)
-    {
-        return str_replace("@", "{{'@'}}", $content);
+        return str_replace(
+            ['__OPEN_CURLIES__', '__CLOSE_CURLIES__', '__OPEN_DANGER__', '__CLOSE_DANGER__', '__LONE_AT__', '__PHP_OPEN__'],
+            ["<?= '{{' ?>", "<?= '}}' ?>", "<?= '{!!' ?>", "<?= '!!}' ?>", "<?= '@' ?>", "<?= '<?php' ?>"],
+            $content
+        );
```

Comment:
I think this would be a little cleaner:

```php
private function getEscapedMarkdownContent($file)
{
    $replacements = ["<?php" => "<{{'?php'}}"];

    if ($file->getFullExtension() == 'md') {
        $replacements = array_merge([
            "@" => "{{'@'}}",
            "{{" => "@{{",
            "{!!" => "@{!!",
        ], $replacements);
    }

    return strtr($file->getContents(), $replacements);
}
```

- we don't need to worry about closing `}}` or `!!}`, since Blade won't care about those if the opening braces aren't present
- using `strtr` allows us to avoid the intermediate `__TOKEN__` dance, since `strtr` won't process a value twice
- since the goal is to set the markdown up to be correctly processed by Blade, I feel like using Blade's `{{` for echoing and `@` for escaping fits better than resorting to `<?= ... ?>`

As for the newline issue, it seems that using Blade to replace the PHP open tag via `<{{'?php'}}` handles newlines correctly, whereas replacing it with `<?= '<?php' ?>` squashes the newline if present. I didn't dig into why, but I figure if the Blade method is parsing it correctly, we might as well stick with that.

Reply:
👌🏻 I'll pull it down and test it. Thanks!